### PR TITLE
Branch

### PR DESCRIPTION
**Pull request by Alex-H1**
Added MyBatis

### CODE REVIEW

File: src/main/java/sql/IOrderDAO.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package sql;
+
+
+import model.Order;
+
+public interface IOrderDAO extends IBaseDAO<Order> {
+
+}
```
Comment (ankochergin) [inline src/main/java/sql/IOrderDAO.java]: add some custom method for orders here

File: src/main/java/sql/IPackageDAO.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package sql;
+
+
+import model.Package;
+
+public interface IPackageDAO extends IBaseDAO<Package> {
+
```
Comment (ankochergin) [inline src/main/java/sql/IPackageDAO.java:7]: add some custom method for package here

File: src/main/java/sql/IPackageTypeDAO.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package sql;
+
+import model.PackageType;
+
+public interface IPackageTypeDAO extends IBaseDAO<PackageType> {
```
Comment (ankochergin) [inline src/main/java/sql/IPackageTypeDAO.java:7]: add some custom method for package type here

**Review by ankochergin — COMMENTED** [review]
also please add some meaningful name for your PR

File: src/main/java/xml/JAXBMain.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            Marshaller marshaller = jaxbContext.createMarshaller();
+            marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+            marshaller.marshal(staff, new File("src/main/resources/delivery_service_output.xml"));
+            LOG.info(JAXBMain.unMarshall());
+        } catch (ParseException | JAXBException e) {
+            LOG.error(e.getMessage());
+        }
+    }
+
+    public static Staff unMarshall() throws JAXBException {
```
Comment (ankochergin) [inline src/main/java/xml/JAXBMain.java]:
```suggestion
    public static Staff unmarshall() throws JAXBException {
```

File: src/main/java/xml/JAXBMain.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            Staff staff = new Staff(1, "Jane", "June", 3, new SimpleDateFormat("dd-MM-yyyy").parse("23-05-2004"));
+            JAXBContext jaxbContext = JAXBContext.newInstance(Staff.class);
+            Marshaller marshaller = jaxbContext.createMarshaller();
+            marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+            marshaller.marshal(staff, new File("src/main/resources/delivery_service_output.xml"));
+            LOG.info(JAXBMain.unMarshall());
+        } catch (ParseException | JAXBException e) {
+            LOG.error(e.getMessage());
+        }
+    }
```
Comment (ankochergin) [inline src/main/java/xml/JAXBMain.java]: please implement jaxb to create the same xml as I shared yesterday

File: src/main/java/model/DeliveryService.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+import javax.xml.bind.annotation.XmlType;
+import java.util.List;
+
+@XmlRootElement(name = "delivery_service")
+@XmlType(propOrder = {"customers", "staff", "companies", "companyTypes", "countries"})
+public class DeliveryService {
+    @JsonProperty
+    private List<Customer> customers;
+    @JsonProperty
+    private List<Staff> Staff;
```
Comment (ankochergin) [inline src/main/java/model/DeliveryService.java]:
```suggestion
    private List<Staff> staff;
```

File: src/main/java/xml/XMLValidator.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import javax.xml.validation.Schema;
+import javax.xml.validation.SchemaFactory;
+import javax.xml.validation.Validator;
+import java.io.File;
+import java.io.IOException;
+
+public class XMLValidator {
+    private static final Logger LOG = LogManager.getLogger(XMLValidator.class);
+
+    public static boolean ValidateXML(String xsdPath, String xmlPath){
```
Comment (ankochergin) [inline src/main/java/xml/XMLValidator.java]:
```suggestion
    public static boolean validateXML(String xsdPath, String xmlPath){
```

**Review by ankochergin — CHANGES_REQUESTED** [review]
1. check that all your variable names and methods have names according to java naming convention
2. rename PR
3. work on formatting

File: src/main/java/JSON/JacksonMain.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
+package JSON;
```
Comment (ankochergin) [inline src/main/java/JSON/JacksonMain.java]: package name must be lowercase

File: src/main/java/sql/IStaffDAO.java [hunk 9]
```diff
@@ -0,0 +1,10 @@
+package sql;
+
+import model.Staff;
+
+import java.sql.SQLException;
+import java.util.List;
+
+public interface IStaffDAO extends IBaseDAO<Staff> {
+    List<Staff> getStaffByName(String name) throws SQLException;
+}
```
Comment (ankochergin) [inline src/main/java/sql/IStaffDAO.java]: please be consistent in formatting. In some classes/interfaces you have empty lines above and after method and in some no empty lines. Please fix like I suggested in all classes

Comment (ankochergin) [inline src/main/java/sql/IStaffDAO.java]:
```suggestion
public interface IStaffDAO extends IBaseDAO<Staff> {

    List<Staff> getStaffByName(String name) throws SQLException;
    
}
```

Comment (ankochergin) [reply src/main/java/sql/IStaffDAO.java]: also please add empty line between variables and methods

**Comment by Alex-H1** [comment]
Refactor and MyBatis progress

**Comment by Alex-H1** [comment]
added mybatis implementation

**Review by ankochergin — APPROVED** [review]

**Comment by Alex-H1** [comment]
added facad and abstract factories

File: src/main/java/Main.java [hunk 10]
```diff
@@ -1,18 +1,47 @@
...
+        customerDAO1.saveEntity(new Customer("Johnny", "A", "Appleseed", "(617)414-4270", 3, 23));
+
+        FacadeFactory factory = new FacadeFactory();
+        IBaseDAO customer = factory.createDAO("customer");
+        List<Customer> customerList1 = customer.getAll();
+        LOG.info(customerList1);
 
+//        IBaseDAO companyDAO1 = factory.createMyBatis("company");
+//        List<Company> companyList1 = companyDAO1.getAll();
+//        LOG.info(companyList1);
```
Comment (ankochergin) [inline src/main/java/Main.java:45]: remove comments if they are not used

File: src/main/java/sql/util/FacadeFactory.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    public IBaseDAO createDAO(String m) throws UnsuportedTypeException {
+        return JDBC.getDAO(m);
+    }
+
+    public IBaseDAO createMyBatis(String m) throws UnsuportedTypeException {
+        return MYBATIS.getDAO(m);
+    }
+
+
+}
```
Comment (ankochergin) [inline src/main/java/sql/util/FacadeFactory.java:24]: I think it's pretty the same as DBFactoryGenerator, could you please think about some other usage of patterns?

**Comment by ankochergin** [comment]
also please don't forget to merge PR's. We should have small PRs for every change

**Pull request merged**